packages/nhsuk-frontend/src/nhsuk/components/file-upload/file-upload.mjs

```diff
@@ -0,0 +1,492 @@
+import { closestAttributeValue } from '../../common/closest-attribute-value.mjs'
+import { ConfigurableComponent } from '../../configurable-component.mjs'
+import { ElementError } from '../../errors/index.mjs'
+import { I18n } from '../../i18n.mjs'
+
+/**
+ * File upload component
+ *
+ * @preserve
+ * @augments ConfigurableComponent<FileUploadConfig>
+ */
+export class FileUpload extends ConfigurableComponent {
+  /**
+   * @private
+   * @type {HTMLFileInputElement}
+   */
+  $input
+
+  /**
+   * @private
+   */
+  $button
+
+  /**
+   * @private
+   */
+  $status
+
+  /** @private */
+  i18n
+
+  /** @private */
+  id
+
+  /** @private */
+  $announcements
+
+  /**
+   * @private
+   * @type {boolean | undefined}
+   */
+  enteredAnotherElement
+
+  /**
+   * @param {Element | null} $root - File input element
+   * @param {FileUploadConfig} [config] - File Upload config
+   */
+  constructor($root, config = {}) {
+    super($root, config)
+
+    const $input = this.$root.querySelector('input')
+
+    if ($input === null) {
+      throw new ElementError({
+        component: FileUpload,
+        identifier: 'File inputs (`<input type="file">`)'
```

Comment:
Then can we pass these options in?

This makes sure `ElementError` automatically switches out the text for both unexpected components types and `null` (not found) values

I've used "Form field" because we're only querying for `input` at this point, we haven't expected a specific type attribute

```suggestion
        element: $input,
        expectedType: 'HTMLInputElement',
        identifier: 'Form field (`<input>`)',
```

Reply:
Fixed.